### PR TITLE
add events for logging in and out

### PR DESCRIPTION
**Pull request by 8633brown**
is this any good for #410

### CODE REVIEW

**Comment by brunogaspar** [comment]
This is nice, just not 100% sure about the event names as of yet, probably just needs some minor rewording and should be good.

Will try to get back to you later today or tomorrow.

Thanks!

File: src/Sentinel.php [hunk 1]
```diff
@@ -486,6 +486,9 @@ public function creatingBasicResponse(Closure $basicResponse)
      */
     public function login(UserInterface $user, $remember = false)
     {
+
+        $this->fireEvent('sentinel.loggingin', $user);
```
Comment (brunogaspar) [inline src/Sentinel.php]: I think we can do something like `sentinel.logging.in` or `sentinel.logging-in`

File: src/Sentinel.php [hunk 2]
```diff
@@ -521,15 +525,19 @@ public function logout(UserInterface $user = null, $everywhere = false)
     {
         $currentUser = $this->check();
 
+        $this->fireEvent('sentinel.loggingout', $user);
```
Comment (brunogaspar) [inline src/Sentinel.php]: Same as above

File: src/Sentinel.php [hunk 3]
```diff
@@ -538,6 +546,7 @@ public function logout(UserInterface $user = null, $everywhere = false)
         $this->persistences->{$method}($user);
 
         $this->user = null;
+        $this->fireEvent('sentinel.loggingout', $user);
```
Comment (brunogaspar) [inline src/Sentinel.php]: Shouldn't this be  a `logged out` event instead?

**Comment by brunogaspar** [comment]
@8633brown Can you please do a rebase on this one to solve the merge conflict and also remove the other unnecessary changes? That way will be easier to merge into master later.

**Comment by 8633brown** [comment]
Sorry. Thought I'd already sorted this. Do you want a PR for master too?

**Comment by brunogaspar** [comment]
2.0 is fine, i can then merge into master after

**Pull request merged**